osu.Game/Graphics/UserInterface/LoadingButton.cs

```diff
@@ -44,18 +40,34 @@ public Vector2 LoadingAnimationSize
 
         protected LoadingButton()
         {
-            AddRange(new[]
+            Add(loading = new LoadingSpinner
             {
-                CreateContent(),
-                loading = new LoadingSpinner
-                {
-                    Anchor = Anchor.Centre,
-                    Origin = Anchor.Centre,
-                    Size = new Vector2(12)
-                }
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Size = new Vector2(12),
+                Depth = -1,
             });
         }
 
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            Add(CreateContent());
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            loading.State.BindValueChanged(s =>
+            {
+                if (s.NewValue == Visibility.Visible)
+                    OnLoadStarted();
+                else
```

Comment:
This is going to run where it potentially didn't before, have you tested all cases?

Reply:
Yeah was aware of that and already checked it, but double-checked deeply just in case and it should not cause any issues.